app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/EarnViewModel.kt

```diff
@@ -153,13 +141,34 @@ class EarnViewModel(val taskRepository: TasksRepository, val wallet: Wallet,
 
     override fun onScreenVisibleToUser() {
         refresh()
-        if (shouldShowTask.get()) {
-            onEarnScreenVisible()
-        } else if (shouldShowNoTask.get()) {
-            onNoTasksAvailableVisible()
-        } else {
-            onLockedScreenVisible()
+        when {
+            shouldShowTask.get() -> onEarnScreenVisible()
+            shouldShowNoTask.get() -> onNoTasksAvailableVisible()
+            else -> onLockedScreenVisible()
         }
+        checkForUpdates()
+    }
+
+    private fun checkForUpdates() {
+        val currentLastUpdatedTime = taskRepository.task?.lastUpdatedAt ?: 0
+        val currentTaskDisplayedId = taskRepository.task?.id.orEmpty()
+
+        taskService.retrieveNextTask(object : OperationCompletionCallback {
+            override fun onError(errorCode: Int) {
+            }
+
+            override fun onSuccess() {
+                if (taskRepository.task != null) {
+                    taskRepository.task?.lastUpdatedAt?.let {
+                        if (it > currentLastUpdatedTime ||
```

Comment:
when will it be false?

Reply:
I am only going to refresh the screen if something has really changed (either this is a completely new task or some of the details of the task have been changed and therefore the last updated time has been changed)